src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Extensions/RangeExtensions.cs

```diff
@@ -144,20 +144,12 @@ static int GetAbsolutePosition(Position position, SourceText sourceText, [Caller
         {
             var line = position.Line;
             var character = position.Character;
-            var lineCount = sourceText.Lines.Count;
-            if (line > lineCount ||
-                (line == lineCount && character > 0))
+            if (!sourceText.TryGetAbsolutePosition(line, character, out var absolutePosition))
```

Comment:
This logic just moved to `SourceTextExtensions`, which is a better place anyway, as we can use it in more places to possibly solve some other PRISM bugs for out of range issues, since it's the only thing that currently does the correct "allow line to be count + 1" logic.